### PR TITLE
#1316 Single Table Inheritance pattern implemented

### PR DESCRIPTION
**Pull request by ved-asole**
Pull request title

- Added the example for the Single Table Inheritance pattern under the PR #1316 

Pull request description

- Added the example for the Single Table Inheritance pattern under the PR #1316 

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

**Comment by ved-asole** [comment]
Hi @iluwatar, the code is running in my system but not sure why it is not working with the checks.. Could you please assist me to resolve this issue?

**Comment by ved-asole** [comment]
Hi @iluwatar, could you please check and confirm if any changes are required?

**Comment by ved-asole** [comment]
Hi @iluwatar, could you please review the above pull request

File: pom.xml [hunk 1]
```diff
@@ -207,6 +207,8 @@
     <module>context-object</module>
     <module>thread-local-storage</module>
     <module>optimistic-offline-lock</module>
+    <module>single-table-inheritance</module>
+    <module>crtp</module>
```
Comment (iluwatar) [inline pom.xml:211]: crtp module should be there in `master` branch already

File: single-table-inheritance/pom.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        <version>1.26.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>single-table-inheritance</artifactId>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
```
Comment (iluwatar) [inline single-table-inheritance/pom.xml:18]: these can be left out

File: single-table-inheritance/src/main/java/com/iluwatar/SingleTableInheritance.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  /**
+   * The starting point of the CommandLineRunner
+   * where the main program is run.
+   *
+   * @param args program runtime arguments
+   */
+  @Override
+  public void run(String... args) throws Exception {
+
+    System.out.println();
```
Comment (iluwatar) [inline single-table-inheritance/src/main/java/com/iluwatar/SingleTableInheritance.java:60]: use a proper logger - from Lombok for example

File: single-table-inheritance/src/main/java/com/iluwatar/entity/TransportVehicle.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  public TransportVehicle(String manufacturer, String model, int loadCapacity) {
+    super(manufacturer, model);
+    this.loadCapacity = loadCapacity;
+  }
+
+  @Override
+  public String toString() {
+    return super.toString();
+  }
```
Comment (iluwatar) [inline single-table-inheritance/src/main/java/com/iluwatar/entity/TransportVehicle.java:28]: seems unnecessary override

**Review by iluwatar — CHANGES_REQUESTED** [review]
Added a few minor remarks. Also, please resolve the conflicts.

File: pom.xml (hunk 1 above)
Comment (ved-asole) [reply pom.xml:211]: sure, I will remove it.

File: single-table-inheritance/pom.xml (hunk 2 above)
Comment (ved-asole) [reply single-table-inheritance/pom.xml:18]: Ok, sure

File: single-table-inheritance/src/main/java/com/iluwatar/SingleTableInheritance.java (hunk 3 above)
Comment (ved-asole) [reply single-table-inheritance/src/main/java/com/iluwatar/SingleTableInheritance.java:60]: Ok

File: single-table-inheritance/src/main/java/com/iluwatar/entity/TransportVehicle.java (hunk 4 above)
Comment (ved-asole) [reply single-table-inheritance/src/main/java/com/iluwatar/entity/TransportVehicle.java:28]: Yes, I think I missed it. Willl remove it

**Comment by ved-asole** [comment]
Hi @iluwatar, I have updated the code as per the review. Please approve the merge request.

**Comment by ved-asole** [comment]
Hi @iluwatar , I have updated the code as per the review. Please approve the merge request.

**Comment by ved-asole** [comment]
Hi @iluwatar, please review the above PR. I have made the necessary changes after the review.

**Comment by ved-asole** [comment]
@iluwatar , could you please review and approve the PR?

**Comment by ved-asole** [comment]
Hi @iluwatar , please approve the PR. I have resolved the merge conflicts

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @ved-asole for code

**Pull request merged**